tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -631,6 +690,10 @@ const std::vector<std::pair<int, int>>& GetTileSizesFrontier() {
   return *frontier;
 }
 
+// Helper structs to help determine which data types to use given the size of
+// the matrix data type. For each size of the data type, we only compile one
+// kernel of that size and every data type sharing the same size will also share
+// the same kernel (unless data alignment issues forbids so).
```

Comment:
Maybe we can replace the second sentence with: "A transpose of elements of size N will use a kernel which operates on an array of TransposeElemType<N>::type."